sound/soc/sof/intel/cnl.c

```diff
@@ -153,6 +153,12 @@ static void cnl_ipc_dsp_done(struct snd_sof_dev *sdev)
 static int cnl_ipc_send_msg(struct snd_sof_dev *sdev,
 			    struct snd_sof_ipc_msg *msg)
 {
+	if (sdev->d0_state == SOF_DSP_D0I3) {
+		snd_sof_dsp_write(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDR,
+				  CNL_DSP_REG_HIPCIDR_BUSY | msg->header);
```

Comment:
This looks wrong. You'd need to have a list of IPCs that use the registers and others that use the windows. we should not allow for generic use of the header for all messages, it's not quite right.